lucene/distribution.tests/src/test/org/apache/lucene/distribution/TestScripts.java

```diff
@@ -0,0 +1,134 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.lucene.distribution;
+
+import com.carrotsearch.procfork.ForkedProcess;
+import com.carrotsearch.procfork.Launcher;
+import com.carrotsearch.procfork.ProcessBuilderLauncher;
+import com.carrotsearch.randomizedtesting.LifecycleScope;
+import com.carrotsearch.randomizedtesting.RandomizedTest;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.Optional;
+import java.util.function.Supplier;
+import org.assertj.core.api.Assertions;
+import org.assertj.core.api.ThrowingConsumer;
+import org.junit.Test;
+
+/** Verify that scripts included in the distribution work. */
+public class TestScripts extends AbstractLuceneDistributionTest {
+  @Test
+  public void testLukeCanBeLaunched() throws Exception {
+    Path distributionPath;
+    if (randomBoolean()) {
+      // Occasionally, be evil: put the distribution in a folder with a space inside. For Uwe.
+      distributionPath = RandomizedTest.newTempDir(LifecycleScope.TEST).resolve("uh oh");
+      Files.createDirectory(distributionPath);
+      new Sync().sync(getDistributionPath(), distributionPath);
+    } else {
+      distributionPath = getDistributionPath();
+    }
+
+    Path lukeScript = resolveScript(distributionPath.resolve("bin").resolve("luke"));
+
+    Launcher launcher =
+        new ProcessBuilderLauncher()
+            .executable(lukeScript)
+            // tweak Windows launcher scripts so that they don't fork asynchronous java.
+            .envvar("DISTRIBUTION_TESTING", "true")
+            .viaShellLauncher()
+            .cwd(distributionPath)
+            .args("--sanity-check");
+
+    execute(
+        launcher,
+        0,
+        (output) -> {
+          Assertions.assertThat(output).contains("[Vader] Hello, Luke.");
+        });
+  }
+
+  /** The value of <code>System.getProperty("os.name")</code>. * */
+  public static final String OS_NAME = System.getProperty("os.name");
+  /** True iff running on Windows. */
+  public static final boolean WINDOWS = OS_NAME.startsWith("Windows");
+
+  protected Path resolveScript(Path scriptPath) {
+    List<Path> candidates = new ArrayList<>();
+    candidates.add(scriptPath);
+
+    String fileName = scriptPath.getFileName().toString();
+    if (WINDOWS) {
+      candidates.add(scriptPath.resolveSibling(fileName + ".cmd"));
+      candidates.add(scriptPath.resolveSibling(fileName + ".bat"));
+    } else {
+      candidates.add(scriptPath.resolveSibling(fileName + ".sh"));
+    }
+
+    Optional<Path> script = candidates.stream().sequential().filter(Files::exists).findFirst();
+    if (script.isEmpty()) {
+      throw new AssertionError("No script found for the base path: " + scriptPath);
+    }
+
+    return script.get();
```

Comment:
This could be more fully functional style with orElseThrow(). But I do not have strong feeling about it.
```suggestion
    return candidates.stream()
        .sequential()
        .filter(Files::exists)
        .findFirst()
        .orElseThrow(() -> new AssertionError("No script found for the base path: " + scriptPath));
```